### DISCUSSION TITLE
@ckeditor/ckeditor5-utils: Insufficient Typings for Parameters for Locale.t()

### DISCUSSION THREAD

**Discussion by mmichaelis** [DefinitelyTyped/DefinitelyTyped#53919] [Issues with a @types package] [answered]
The typings for `Locale.t()` don't match the specification and usage:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7740c7360101ec9bb93b37ea2d160c600822e7f9/types/ckeditor__ckeditor5-utils/src/locale.d.ts#L78

* `message` may not only be a string but also of type [Message](https://ckeditor.com/docs/ckeditor5/28.0.0/api/module_utils_translation-service-Message.html).

* Optional parameter `values` may not only be `string[]` but according to [documentation](https://ckeditor.com/docs/ckeditor5/28.0.0/api/module_utils_locale-Locale.html#function-t):

    ```text
    String | Number | Array.<(String | Number)>
    ```

Suggested typing:

```
t(message: string|Message, values?: string|number|(string|number)[]): string;
```

**Comment by fedemp** [comment, marked as answer]
@mmichaelis This has been fixed in `master`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/905355a9745d99e7c2871bd6d3b4b7cbe2a78672/types/ckeditor__ckeditor5-utils/src/locale.d.ts#L84

Can you mark this as answered? :+1:
